ray-curator/ray_curator/stages/classifiers/base.py

```diff
@@ -0,0 +1,479 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
naming nit.. should these be stages/text_models/base.py + stages/text_models/classifiers/{aegis, domain}.py + stages/text_models/embedding_model.py?

Given even SemDedup's embedding generation looks very similar to this (and it need not live inside semdedup but rather here)..

cc @ayushdg
cc @abhinavg4 @suiyoubi (since a similar situation) likely to happen in video too

i don't have a strong opinion

Reply:
Irresepctive of what we decide can you seee if the current code will also be able to support the embedding generation model?

Reply:
Sure, I can try to split up into a more generic structure.